openwisp_monitoring/device/static/monitoring/js/device-map.js

```diff
@@ -236,22 +236,35 @@
                     let bounds = event.target.getBounds();
                     if (bounds._southWest.lng < -180 && !netjsonGraph.westWorldFeaturesAppended) {
                         let westWorldFeatures = window.structuredClone(netjsonGraph.data);
+                        // Exclude the features that may be added for the East world map
+                        westWorldFeatures.features = westWorldFeatures.features.filter(
+                            element => element.geometry.coordinates[0] <= 180
+                        )
                         westWorldFeatures.features.forEach(element => {
                             if (element.geometry) {
                                 element.geometry.coordinates[0] -= 360;
                             }
                         });
+                        // netjsonGraph.utils.appendData call the render method which
+                        // super imposes the data on the existing points on the map.
+                        netjsonGraph.leaflet.geoJSON.removeFrom(netjsonGraph.leaflet);
```

Comment:
The points were getting duplicated by just using appendData. I reviewed the code of netjsongraph.js and appendData do not call `netjsonGraph.leaflet.geoJSON.removeFrom(netjsonGraph.leaflet);` like `overrideData` does. 

https://github.com/openwisp/netjsongraph.js/blob/90799710ba2eec7d440a67e58d98452ce3f28aec/src/js/netjsongraph.update.js#L176-L184

I may be wrong, but we may need to replicate this in appendData as well.

Reply:
I did not understand what you mean exactly with:

> I may be wrong, but we may need to replicate this in appendData as well.

You mean that we should patch that method in netjsongraph.js, or what?